src/util/UriUtil.ts

```diff
@@ -0,0 +1,51 @@
+import { DataFactory } from 'n3';
+import type { Literal, NamedNode, Term } from 'rdf-js';
+import { CONTENT_TYPE } from './UriConstants';
+
+// Shorthands for commonly used predicates
+const shorthands: { [id: string]: NamedNode } = {
+  contentType: DataFactory.namedNode(CONTENT_TYPE),
+};
+
+// Caches named node conversions
+const termMap: { [id: string]: NamedNode } = {};
+
+/**
+ * @param input - Checks if this is a {@link Term}.
+ */
+export const isTerm = (input?: any): input is Term => input?.termType;
+
+/**
+ * Converts the incoming name to a named node if needed.
+ * In case of string, first checks if it is a shorthand, if not a new named node gets made.
+ * The generated terms get cached to prevent the amount of named nodes that get created,
+ * so only use this for internal constants!
+ * @param name - Predicate to potentially transform.
+ */
+export const getNamedNode = (name: NamedNode | string): NamedNode => {
+  if (typeof name === 'string') {
+    if (shorthands[name]) {
+      return shorthands[name];
+    }
+    if (!termMap[name]) {
+      termMap[name] = DataFactory.namedNode(name);
```

Comment:
We probably want to guard here against a DDOS attack where our named node map is filled with spam.
For now, it could just be easy to periodically empty the object.

Reply:
The function currently only gets used on our constants in the entire codebase, so the size of this map can't increase above the amount of constants we have (until someone writes a module that breaks this rule ofc). For incoming subject URIs a separate call to `namedNode` still happens. I was thinking about potentially using some sort of cache that keeps the X most recent entries so it could also be used for subjects.

Reply:
Excellent; agree on LRU if ever needed.